### PR TITLE
[15.0] [MIG] account_payment_partner

### PR DESCRIPTION
**Pull request by misern2**
Standard Migration
Depends on
- [x] #870

### CODE REVIEW

File: account_payment_partner/models/account_move.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+                move.partner_bank_filter_type_domain = move.bank_partner_id
+            elif move.move_type in ("in_invoice", "out_refund"):
+                move.partner_bank_filter_type_domain = move.commercial_partner_id
+            else:
+                move.partner_bank_filter_type_domain = False
+
+    @api.depends("partner_id", "company_id")
+    def _compute_payment_mode(self):
+        for move in self:
+            move.payment_mode_id = move.payment_mode_id
```
Comment (i-vyshnevska) [inline account_payment_partner/models/account_move.py]:
```suggestion
```

File: account_payment_partner/models/account_move.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    @api.depends("partner_id", "payment_mode_id")
+    def _compute_partner_bank(self):
+        for move in self:
+            # No bank account assignation is done for out_invoice as this is only
+            # needed for printing purposes and it can conflict with
+            # SEPA direct debit payments. Current report prints it.
+            def get_bank_id():
+                return move.commercial_partner_id.bank_ids.filtered(
+                    lambda b: b.company_id == move.company_id or not b.company_id
+                )[:1]
```
Comment (i-vyshnevska) [inline account_payment_partner/models/account_move.py]:
```suggestion
                return fields.first(move.commercial_partner_id.bank_ids.filtered(
                    lambda b: b.company_id == move.company_id or not b.company_id
                ))
```

**Comment by simahawk** [comment]
@misern2 to depend on #870 for the build, you can add a `test-requirements.txt` file (if not present yet) and add a line like this
```
odoo${10|11|12|13|14}-addon-${addon_name} @ git+https://github.com/OCA/$repo_name@refs/pull/$PR_NR/head#subdirectory=setup/$addon_name
```
In this case it would be:
```
odoo-addon-account_payment_mode @ git+https://github.com/OCA/bank-payment@refs/pull/870/head#subdirectory=setup/account_payment_mode
```
Note that for v15 we don't need the Odoo version in the pkg namespace `odoo-addon`.

File: account_payment_partner/models/account_move.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+    payment_mode_filter_type_domain = fields.Char(
+        compute="_compute_payment_mode_filter_type_domain"
+    )
+    partner_bank_filter_type_domain = fields.Many2one(
+        comodel_name="res.partner", compute="_compute_partner_bank_filter_type_domain"
+    )
+    payment_mode_id = fields.Many2one(
+        comodel_name="account.payment.mode",
+        compute="_compute_payment_mode",
```
Comment (simahawk) [inline account_payment_partner/models/account_move.py]:
```suggestion
        compute="_compute_payment_mode_id",
```

File: account_payment_partner/models/account_move.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+                or self.partner_id.valid_mandate_id.partner_bank_id
+            )
+        # Return this as empty recordset
+        return self.partner_bank_id
+
+    @api.model
+    def create(self, vals):
+        """Force compute partner_bank_id when invoice is created from SO
+        to avoid that odoo _prepare_invoice method value will be set.
+        """
```
Comment (simahawk) [inline account_payment_partner/models/account_move.py]: this is not the docstring of the create method, use a comment pls

File: account_payment_partner/tests/test_account_payment_partner.py [hunk 5]
```diff
@@ -0,0 +1,580 @@
...
+from odoo import _, fields
+from odoo.exceptions import UserError, ValidationError
+from odoo.fields import Date
+from odoo.tests.common import Form, TransactionCase
+
+
+class TestAccountPaymentPartner(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
```
Comment (simahawk) [inline account_payment_partner/tests/test_account_payment_partner.py:14]:
```suggestion
        super().setUpClass()
        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
```

File: account_payment_partner/models/account_move.py [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+        store=True,
+        ondelete="restrict",
+        readonly=False,
+        check_company=True,
+    )
+    bank_account_required = fields.Boolean(
+        related="payment_mode_id.payment_method_id.bank_account_required", readonly=True
+    )
+    partner_bank_id = fields.Many2one(
+        compute="_compute_partner_bank",
```
Comment (simahawk) [inline account_payment_partner/models/account_move.py]:
```suggestion
        compute="_compute_partner_bank_id",
```

**Review by simahawk — CHANGES_REQUESTED** [review]
LG overall

File: account_payment_partner/models/account_move.py (hunk 2 above)
Comment (misern2) [reply account_payment_partner/models/account_move.py]: @i-vyshnevska Thanks for pointing out. I didn't know this method! :+1:

**Comment by alexis-via** [comment]
No need to add the field commercial_partner_id as invisible to the "view_move_form" view because it is already present in the original view, cf:
https://github.com/odoo/odoo/blob/15.0/addons/account/views/account_move_views.xml#L733

**Comment by misern2** [comment]
> No need to add the field commercial_partner_id as invisible to the "view_move_form" view because it is already present in the original view, cf: https://github.com/odoo/odoo/blob/15.0/addons/account/views/account_move_views.xml#L733

@alexis-via  Done!. :+1:

**Review by eduardgm — APPROVED** [review]
Functional review, LGTM +1

**Review by HaraldPanten — APPROVED** [review]
Functional review 👍

@simahawk Could you update your review, please?

@alexis-via @pedrobaeza WDYT about this PR? Could it be merged?

TY all.

**Pull request merged**